### PR TITLE
Watch rows/cols and resize the term and send sigwinch signal to exec

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

**Comment by jwforres** [comment]
@ncdc try this with your change to the API, if you update the rows/cols in the example index file it should resize the term and send the signal

@stefwalter decided to make it up to whoever is using the container terminal to decide when / if to resize it by updating the rows cols we pass into the scope

**Comment by jwforres** [comment]
updated, fixed a bug

**Comment by stefwalter** [comment]
Nice work.

Is there a link to the kubernetes pull request to test this against? I'd like to test this before merging.

**Comment by ncdc** [comment]
I'm still working on it - the WIP is here: https://github.com/ncdc/kubernetes/pull/7

**Comment by ncdc** [comment]
And yes, I think waiting for the kube PR to merge makes sense

File: dist/container-terminal.js [hunk 1]
```diff
@@ -213,6 +215,15 @@
                             alive = null;
                         }
 
+                        scope.$watchGroup(["cols", "rows"], function(newValues) {
+                          var cols = newValues[0] || defaultCols;
+                          var rows = newValues[1] || defaultRows;
+                          term.resize(cols, rows);
+                          if (ws && ws.readyState === 1) {
+                            ws.send("4" + window.btoa('{"Width": cols, "Height": rows}'));
```
Comment (ncdc) [inline dist/container-terminal.js]: Need to use the actual values for cols and rows 😄 

``` javascript
ws.send("4" + window.btoa('{"Width":' + cols + ',"Height":' + rows + '}'));
```

Comment (jwforres) [inline dist/container-terminal.js]: Hah whoops, good catch. Will update that in the morning

Comment (ncdc) [inline dist/container-terminal.js]: Cool. With this change, it works!

**Comment by ncdc** [comment]
Kube PR: https://github.com/kubernetes/kubernetes/pull/25273

**Comment by jwforres** [comment]
@ncdc saw the API PR is trying to get merged, did anything change or is the impl in this PR still correct for sending the signal

**Comment by ncdc** [comment]
@jwforres it should still be the same but I'll doublecheck to make sure this still works

**Comment by ncdc** [comment]
@jwforres yes it works, but you still need to do https://github.com/kubernetes-ui/container-terminal/pull/7#discussion_r61191752

**Comment by jwforres** [comment]
rebased and fixed https://github.com/kubernetes-ui/container-terminal/pull/7#discussion_r61191752

should be ready to go as soon as the API change gets merged

**Comment by jwforres** [comment]
cc @spadgett

**Pull request merged**